stan-dev/stan
# Feature/2418 manual ref html

#### Submission Checklist


#### Summary

Split current manual into three parts:

* reference manual (language, algorithms, reproducibility)
* functions reference
* user's guide (examples, programming guide)

The reference manual is converted to bookdown and generates HTML.  The longer-term plan is to convert the functions reference and user's guide in the same way and maybe move the functions reference into the reference manual.

I updated names for 2.18, but this will have to be updated when 2.18 is actually released to build in the right doc names.

#### Intended Effect

Provide HTML manuals, modularize the manual into functional components, and make the manuals easier for everyone to update.

#### How to Verify

```
> cd stan-dev/stan
> make doc
```

This will make all of the doc in a new directory `doc/`, the contents of which include

* reference manual:  html and pdf
* functions reference: pdf
* functions reference: .txt for R applications
* users guide: pdf

We discussed whether to number the web site and decided it would just mess up indexing given that we don't want to support old versions.  The pdf documents are version numbered.

To deploy, the link from the html to the pdf reference manual, the index.html (and all other .html files) for the HTML version should go in the same directory as the pdf.

I tried to turn the toolbar off (the thing that lets you change fonts, click for the PDF, and has a redundant title), but that doesn't work in any of the browsers.  The toolbar is empty other than title in Safari for me and the toolbar is populated in Chrome (with things that work, modulo the link comment above).

#### Side Effects

Removed `make manual` target in favor of `make doc`.

Markup for changes is removed.

#### Documentation

Affirmative.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
